src/crystal/system/win32/file_descriptor.cr

```diff
@@ -276,6 +278,70 @@ module Crystal::System::FileDescriptor
   end
 end
 
+private module ConsoleUtils
+  # N UTF-16 code units correspond to no more than 3*N UTF-8 code units.
+  private BUFFER_SIZE = 64
```

Comment:
**question:** Is there any particular reason for such a small buffer size? As mentioned in https://github.com/crystal-lang/crystal/pull/13758#pullrequestreview-1663946275 I'd probably expect this to be a bit bigger. I suppose even a console can send large amounts of data on stdin. Increasing the buffer should yield more performance for negligable cost.

Reply:
If a large buffer is used, static arrays cannot be used, and the data state can only be stored on class variables.
A small buffer is usually sufficient, even when encountering situations where a large amount of data is input from the console, completing data reads multiple times will not become a performance bottleneck compared to human input speed.

Reply:
I'm not sure if a Windows console is necessarily always interactive input. But even if, it's not just a human typing. You could paste a big chunk of data, for example. It seems unnecessarily restrictive to read in such minuscle steps.

What's the problem with a bigger `StaticArray`? Of course we wouldn't want to put *too* much on the stack to avoid running out of memory. But Crystal's fiber stacks are 8MB. It's totally reasonable to use bigger buffers. `IO::DEFAULT_BUFFER_SIZE` is 32KB and it's used all over the place for buffers on the stack.
We might not want exactly `IO::DEFAULT_BUFFER_SIZE` considering the cited comment about size limitation at 10KB from https://github.com/crystal-lang/crystal/pull/13758#pullrequestreview-1663946275.